src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -2291,6 +2291,18 @@ public static SyntaxKind GetPredefinedKeywordKind(SpecialType specialType)
 
         public static SimpleNameSyntax GetRightmostName(this ExpressionSyntax node)
         {
+            var awaitExpression = node as AwaitExpressionSyntax;
+            if (awaitExpression != null && awaitExpression.Expression != null)
+            {
```

Comment:
We're going far out of ht realm of what this function was for.  this was for:  I have a dotted name.  I want the rightmost name.